File: javalin6/javalin-testing-example/src/main/java/io/javalin/example/java/JavalinTestingExampleApp.java
package io.javalin.example.java;

import io.javalin.Javalin;
import static io.javalin.apibuilder.ApiBuilder.get;
import static io.javalin.apibuilder.ApiBuilder.post;

public class JavalinTestingExampleApp {

    private Javalin app;

    public JavalinTestingExampleApp(String dependency) {
        app = Javalin.create(config -> {
            config.router.apiBuilder(() -> {
                get("/users", UserController::getAll);
                post("/users", UserController::create);
                get("/ui", ctx -> ctx.html("<h1>User UI</h1>"));
            });
        });
    }

    public Javalin javalinApp() {
        return app;
    }

}

File: javalin7/javalin-testing-example/src/main/kotlin/io/javalin/example/kotlin/JavalinTestingExampleApp.kt
package io.javalin.example.kotlin

import io.javalin.Javalin
import io.javalin.apibuilder.ApiBuilder.get
import io.javalin.apibuilder.ApiBuilder.post

class JavalinTestingExampleApp(dependency: String) {

    val app = Javalin.create {
        it.router.apiBuilder {
            get("/users", UserController::getAll)
            post("/users", UserController::create)
            get("/ui") { it.html("<h1>User UI</h1>") }
        }
    }

}
